### PR TITLE
[MNG-7571] add Maven API javadoc

### PR DESCRIPTION
**Pull request by hboutemy**
https://issues.apache.org/jira/browse/MNG-7571

generated site with this PR:
https://maven.apache.org/ref/4-LATEST/maven-api/maven-api-settings/index.html
https://maven.apache.org/ref/4-LATEST/maven-api/maven-api-model/index.html
https://maven.apache.org/ref/4-LATEST/maven-settings/index.html
https://maven.apache.org/ref/4-LATEST/maven-model/index.html

### CODE REVIEW

File: api/maven-api-core/src/site/site.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline api/maven-api-core/src/site/site.xml:30]: Canonically just 'Javadoc'

File: api/maven-api-meta/pom.xml [hunk 2]
```diff
@@ -27,6 +27,7 @@
   </parent>
 
   <artifactId>maven-api-meta</artifactId>
-  <name>Maven API Meta annotations</name>
+  <name>Maven 4 API Meta annotations</name>
+  <description>Java 5 annotations for Maven 4 Immutable API.</description>
```
Comment (michael-o) [inline api/maven-api-meta/pom.xml:31]: I think the 5 is useless there days. Just "Java annotations" or "annotations" are good enough.

File: api/maven-api-meta/pom.xml [hunk 3]
```diff
@@ -27,6 +27,7 @@
   </parent>
 
   <artifactId>maven-api-meta</artifactId>
-  <name>Maven API Meta annotations</name>
+  <name>Maven 4 API Meta annotations</name>
```
Comment (michael-o) [inline api/maven-api-meta/pom.xml:30]: Annotations

File: api/maven-api-meta/src/site/site.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline api/maven-api-meta/src/site/site.xml:30]: ditto

File: api/maven-api-model/pom.xml [hunk 5]
```diff
@@ -28,8 +28,8 @@ under the License.
 
   <artifactId>maven-api-model</artifactId>
 
-  <name>Maven API Model</name>
-  <description>Maven API Model for Maven POM (Project Object Model)</description>
+  <name>Maven 4 API - Model</name>
+  <description>Maven 4 API - immutable Model for Maven POM (Project Object Model)</description>
```
Comment (michael-o) [inline api/maven-api-model/pom.xml:32]: Immutable

File: api/maven-api-model/src/site/apt/index.apt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ Introduction
+ -----
+ Jason van Zyl
+ Vincent Siveton
+ Hervé Boutemy
+ -----
+ 2011-06-12
+ -----
+
+Maven 4 API - immutable Maven Model
```
Comment (michael-o) [inline api/maven-api-model/src/site/apt/index.apt:28]: &ndash; Immutable

File: api/maven-api-model/src/site/site.xml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline api/maven-api-model/src/site/site.xml:30]: ditto

File: api/maven-api-settings/pom.xml [hunk 8]
```diff
@@ -28,8 +28,8 @@ under the License.
 
   <artifactId>maven-api-settings</artifactId>
 
-  <name>Maven API Settings</name>
-  <description>Maven API Settings model.</description>
+  <name>Maven 4 API - Settings</name>
```
Comment (michael-o) [inline api/maven-api-settings/pom.xml:31]: So this module contains settings or the settings model? This is a difference.

File: api/maven-api-settings/src/site/apt/index.apt [hunk 9]
```diff
@@ -23,14 +23,13 @@
  2006-11-04
  -----
 
-Maven Settings Model
+Maven 4 API - immutable Settings Model
```
Comment (michael-o) [inline api/maven-api-settings/src/site/apt/index.apt:26]: ditto

File: api/maven-api-toolchain/pom.xml [hunk 10]
```diff
@@ -27,8 +27,8 @@ under the License.
 
   <artifactId>maven-api-toolchain</artifactId>
 
-  <name>Maven API Toolchain</name>
-  <description>Maven API Toolchain model.</description>
+  <name>Maven 4 API - Toolchain</name>
```
Comment (michael-o) [inline api/maven-api-toolchain/pom.xml:30]: ditto

File: api/maven-api-toolchain/src/site/apt/index.apt [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+ -----
+ Introduction
+ -----
+ Vincent Siveton
+ -----
+ 2006-11-04
+ -----
+
+Maven 4 API - immutable Toolchains Model
```
Comment (michael-o) [inline api/maven-api-toolchain/src/site/apt/index.apt:26]: ditto

File: api/maven-api-toolchain/src/site/site.xml [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline api/maven-api-toolchain/src/site/site.xml:30]: ditto

File: api/maven-api-xml/pom.xml [hunk 13]
```diff
@@ -27,7 +27,8 @@
   </parent>
 
   <artifactId>maven-api-xml</artifactId>
-  <name>Maven API XML</name>
+  <name>Maven 4 API XML</name>
```
Comment (michael-o) [inline api/maven-api-xml/pom.xml:30]: I think both name and artifactId are too generics. `maven-api-xml-helpers`, etc. fully describe the intent

File: api/maven-api-xml/src/site/site.xml [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline api/maven-api-xml/src/site/site.xml:30]: ditto

File: api/modello-plugin-velocity/src/site/site.xml [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline api/modello-plugin-velocity/src/site/site.xml:30]: ditto

File: api/src/site/site.xml [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline api/src/site/site.xml:30]: ditto

File: maven-toolchain-builder/src/site/site.xml [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline maven-toolchain-builder/src/site/site.xml:30]: ditto

File: maven-toolchain-model/src/site/site.xml [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+
+<project xmlns="http://maven.apache.org/DECORATION/1.8.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/DECORATION/1.8.0 http://maven.apache.org/xsd/decoration-1.8.0.xsd">
+
+  <edit>${project.scm.url}</edit>
+
+  <body>
+    <menu name="Overview">
+      <item name="Introduction" href="index.html"/>
+      <item name="JavaDocs" href="apidocs/index.html"/>
```
Comment (michael-o) [inline maven-toolchain-model/src/site/site.xml:30]: ditto

File: maven-xml-impl/pom.xml [hunk 19]
```diff
@@ -26,8 +26,8 @@ under the License.
   </parent>
 
   <artifactId>maven-xml-impl</artifactId>
-
-  <name>Maven XML Implementation</name>
+  <name>Implementation of Maven API XML</name>
```
Comment (michael-o) [inline maven-xml-impl/pom.xml:29]: see above

File: plexus-utils/pom.xml [hunk 20]
```diff
@@ -28,6 +28,8 @@ under the License.
   <groupId>org.apache.maven</groupId>
   <artifactId>plexus-utils</artifactId>
   <version>4.0.0-alpha-4-SNAPSHOT</version>
+  <name>Apache Maven Plexus-Utils</name>
+  <description>Apache Maven repackaging of Plexus Utils with immutable Dom interface and its replacement implementation.</description>
```
Comment (michael-o) [inline plexus-utils/pom.xml:32]: DOM

**Review by michael-o — CHANGES_REQUESTED** [review]
Mostly stylistic issues.

An general issue on separation: `<name>Maven 4 API - Toolchain</name>`

Stylistically a hyphen is the wrong symbol. It should be an en dash in this case, but many of our components use `::`. See Doxia, Wagon, Core ITs, Indexer. So I would make it consistent here: `::`

File: plexus-utils/pom.xml (hunk 20 above)
Comment (hboutemy) [reply plexus-utils/pom.xml:32]: no, the interface is named `Dom`, which is not the best name IMHO...

File: api/maven-api-core/src/site/site.xml (hunk 1 above)
Comment (hboutemy) [reply api/maven-api-core/src/site/site.xml:30]: we'll need to fix it in the whole Maven core project...
see https://maven.apache.org/ref/4.0.0-alpha-3/ or https://maven.apache.org/ref/3.8.6/

File: api/maven-api-meta/pom.xml (hunk 3 above)
Comment (hboutemy) [reply api/maven-api-meta/pom.xml:30]: should we rename the artifactId?

File: plexus-utils/pom.xml (hunk 20 above)
Comment (michael-o) [reply plexus-utils/pom.xml:32]: Really? This is confusing, it thought it is an acronym referring to Document Object Model...

File: api/maven-api-core/src/site/site.xml (hunk 1 above)
Comment (michael-o) [reply api/maven-api-core/src/site/site.xml:30]: Yes, it is everywhere :-(

File: api/maven-api-settings/pom.xml (hunk 8 above)
Comment (hboutemy) [reply api/maven-api-settings/pom.xml:31]: I don't get

File: api/maven-api-meta/pom.xml (hunk 3 above)
Comment (michael-o) [reply api/maven-api-meta/pom.xml:30]: Yes, good idea: `maven-api-meta-annotations`?

File: api/maven-api-settings/pom.xml (hunk 8 above)
Comment (michael-o) [reply api/maven-api-settings/pom.xml]: Settings is something you configure/set on something, a settings module is the transport class which is used to plug the actual settings between a user and the backend.

Looking at the code it looks like Maven 4 API :: Settings Utils/Helpers. I don't see any settings nor a model.

Comment (gnodet) [reply api/maven-api-settings/pom.xml]: The jar includes the generated model too:
  https://maven.apache.org/ref/4-LATEST/apidocs/org/apache/maven/api/settings/package-summary.html

File: api/maven-api-xml/pom.xml (hunk 13 above)
Comment (gnodet) [reply api/maven-api-xml/pom.xml:30]: It's not helpers, it contains a single class which is the XML API used by maven:
  https://maven.apache.org/ref/4-LATEST/apidocs/org/apache/maven/api/xml/package-summary.html

File: plexus-utils/pom.xml (hunk 20 above)
Comment (gnodet) [reply plexus-utils/pom.xml:32]: The name comes from the plexus class [`Xpp3Dom`](https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/xml/Xpp3Dom.java) but I removed the `Xpp3` part.  And yes, it's referring to Document Object Model, but I would not call it `DOM`. We could rename it `XmlNode` ?

File: maven-xml-impl/pom.xml (hunk 19 above)
Comment (gnodet) [reply maven-xml-impl/pom.xml:29]: This module actually contains the implementation of the XML `Dom` interface defined by the API.

File: plexus-utils/pom.xml (hunk 20 above)
Comment (michael-o) [reply plexus-utils/pom.xml:32]: Sounds reasonable

**Comment by hboutemy** [comment]
I'll merge this PR regarding basic documentation improvements

on the renaming ideas (`Dom` interface to `XmlNode`, `maven-api-meta` to `maven-api-annotations`), we'll do that separately while doing the plan discussed on dev ML https://lists.apache.org/thread/t52x7jfqlfxwtftbwv4vpfhqg6zgsysf

**Comment by jira-importer** [comment]
Resolve #8881

**Pull request merged**